peachjam/admin.py

```diff
@@ -413,8 +414,19 @@ class NewForm(self.new_document_form_mixin, form):
         return super().get_form(request, obj, **kwargs)
 
     def save_model(self, request, obj, form, change):
+        from peachjam.models import citations_processor
+
         if not change:
             obj.created_by = request.user
+
+        # if date, title, or content changed, re-extract citations or if new document
+        if (
+            not change
+            or ["date", "title", "alternative_name", "citation"] in form.changed_data
```

Comment:
@longhotsummer It seems we can't  get changed_data for `alternative_names` here but we can get it in the save_related method. Should we add this logic there as well?

Reply:
Responding to changes in `alternative_names` is essential, so we have to do it somehow.